discovery/src/main/java/org/wildfly/extension/discovery/StaticProviderDefinition.java

```diff
@@ -0,0 +1,82 @@
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.discovery;
+
+import static org.wildfly.extension.discovery.DiscoveryExtension.DISCOVERY_PROVIDER_RUNTIME_CAPABILITY;
+import static org.wildfly.extension.discovery.DiscoveryExtension.STATIC_PROVIDER;
+
+import org.jboss.as.controller.ObjectListAttributeDefinition;
+import org.jboss.as.controller.ObjectTypeAttributeDefinition;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.ResourceDefinition;
+import org.jboss.as.controller.SimpleAttributeDefinition;
+import org.jboss.as.controller.SimpleResourceDefinition;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.as.controller.registry.OperationEntry;
+import org.jboss.dmr.ModelType;
+
+/**
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ */
+final class StaticProviderDefinition extends SimpleResourceDefinition {
+
+    static final SimpleAttributeDefinition ABSTRACT_TYPE = new SimpleAttributeDefinition(DiscoveryExtension.ABSTRACT_TYPE, ModelType.STRING, true);
+    static final SimpleAttributeDefinition ABSTRACT_TYPE_AUTHORITY = new SimpleAttributeDefinition(DiscoveryExtension.ABSTRACT_TYPE_AUTHORITY, ModelType.STRING, true);
+    static final SimpleAttributeDefinition URI = new SimpleAttributeDefinition(DiscoveryExtension.URI, ModelType.STRING, false);
+    static final SimpleAttributeDefinition URI_SCHEME_AUTHORITY = new SimpleAttributeDefinition(DiscoveryExtension.URI_SCHEME_AUTHORITY, ModelType.STRING, true);
+
+    static final SimpleAttributeDefinition NAME = new SimpleAttributeDefinition(DiscoveryExtension.NAME, ModelType.STRING, false);
+    static final SimpleAttributeDefinition VALUE = new SimpleAttributeDefinition(DiscoveryExtension.VALUE, ModelType.STRING, true);
+
+    static final ObjectTypeAttributeDefinition ATTRIBUTE = new ObjectTypeAttributeDefinition.Builder(DiscoveryExtension.ATTRIBUTE, NAME, VALUE).build();
+
+    static final ObjectListAttributeDefinition ATTRIBUTES = new ObjectListAttributeDefinition.Builder(DiscoveryExtension.ATTRIBUTES, ATTRIBUTE).build();
```

Comment:
Most places we call these kinds of things "properties". Calling them 'attributes' is ok if people are going to learn that term in other contexts or if it's just important somehow, but otherwise being like other subsystems is good too.